chirps_utility.py

```diff
@@ -0,0 +1,140 @@
+from chirps import *
+from main import *
+from csv_utility import *
+from user_utility import *
+from random_id_generator import *
+
+class ChirpsUtility:
+
+	def view_chirps(public_convos, private_convos, chirps_dict, current_user):
+		"""
+		Prints out all public and private chirps that the current user is a part of
+
+		Arg- chirps dictionary (from CSV file), currently selected user
+		"""
+		# counter and empty list to be used for convo select
+		# (display counter as choice number, append each convo to convo_id_list,
+		# convo int(counter - 1) to find convo in convo_id_list by index
+		counter = 1
+		convo_id_list = list()
+		# if a user is signed in
+		if current_user != None:
+			try:
+				print("\n ~ PRIVATE CHIRPS ~ ")
+				for convo_key, chirps_list in private_convos.items():
+					# chirp info that started convo
+					convo_start = chirps_list[0]
+					# if current user is the recipient of a private chirp
+					if chirps_dict[convo_start][1] == current_user.screen_name:
+						print("{0}. {1}: {2}".format(counter, chirps_dict[convo_start][0], chirps_dict[convo_start][2]))
+						convo_id_list.append(convo_key, "private")
+						counter += 1
+					# if current user sent a private chirp
+					elif chirps_dict[convo_start][0] == current_user.screen_name:
+						print("{0}. to {1}: {2}".format(counter, chirps_dict[convo_start][1], chirps_dict[convo_start][2]))
+						convo_id_list.append([convo_key, "private"])
+						counter += 1
+				# print all public chirps
+				return ChirpsUtility.print_public_chirps(public_convos, convo_id_list, chirps_dict, counter)
+			except AttributeError:
+				return ChirpsUtility.print_public_chirps(public_convos, convo_id_list, chirps_dict, counter)
+			except KeyError:
+				return ChirpsUtility.print_public_chirps(public_convos, convo_id_list, chirps_dict, counter)
+		else:
+			return ChirpsUtility.print_public_chirps(public_convos, convo_id_list, chirps_dict, counter)
+
+	def print_public_chirps(public_convos, convo_id_list, chirps_dict, counter):
+		"""
+		Loops through chirps dictionary and prints out each one
+		Takes user input and sends back chirp id for the selected chirp
+
+		Args- chirps dictionary, counter value post private chirps
+		"""
+		print("\n ~ PUBLIC CHIRPS ~ ")
+		for convo_key, chirps_list in public_convos.items():
+			# chirp info that started convo
+			convo_start = chirps_list[0]
+			print("{0}. {1}: {2}".format(counter, chirps_dict[convo_start][0], chirps_dict[convo_start][1]))
+			convo_id_list.append([convo_key, "public"])
+			counter += 1
+		print("\nSelect a chirp, or enter 'q' to quit")
+		convo = input("> ")
+		# select convo based on user input
+		try:
+			if int(convo) <= counter:
+				selected_convo_id = convo_id_list[int(convo) - 1]
+				return selected_convo_id
+			elif convo == "q":
+				show_menu()
```

Comment:
You're on the right track here, but your order of statements should be switched, with the test for `q` coming first.  As it is, if `q` is typed, the `int()` conversion errors out, which is then caught [here](https://github.com/nss-day-cohort-13/birdyboard-josephkane/pull/1/files?diff=unified#diff-5bc02cefb3ea9e27f1a6776eabd1935dR139), which _does_ get you back to the main menu, but results in some weird output:
```
Select a chirp, or enter 'q' to quit
> q
invalid literal for int() with base 10: 'q'

Please enter a valid choice.
```